### PR TITLE
Improve implementations, code style, and tests

### PR DESCRIPTION
**Pull request by turbolent**
- Use the latest version of the Flow CLI, the Cadence 1.0 preview release is outdated
- Make the code more consistent, by fixing whitespace, improving argument labels, removing unnecessary type annotations, etc. 
- Replace implementations with functions (e.g. `String.contains`) and functionality (e.g. `InclusiveRange`) built into Cadence now. In some cases this makes the code much more efficient, e.g. `InclusiveRange` does not allocate a whole array of indices like before
- Extend the test cases

### CODE REVIEW

File: contracts/ArrayUtils.cdc [hunk 1]
```diff
@@ -19,50 +20,63 @@ access(all) contract ArrayUtils {
 
     access(all) fun reverse(_ array: [Int]): [Int] {
         var res: [Int] = []
-        var i: Int = array.length - 1
-        while i >= 0 {
-            res.append(array[i])
-            i = i - 1
+        let length = array.length
```
Comment (bluesign) [inline contracts/ArrayUtils.cdc:23]: maybe `return array.reverse()`?

File: contracts/ArrayUtils.cdc [hunk 2]
```diff
@@ -19,50 +20,63 @@ access(all) contract ArrayUtils {
...
 
-    access(all) fun map(_ array: [AnyStruct], _ f : fun (AnyStruct): AnyStruct) : [AnyStruct] {
-        var res : [AnyStruct] = []
-        for item in array{
+    access(all) fun map(_ array: [AnyStruct], _ f: fun (AnyStruct): AnyStruct): [AnyStruct] {
+        var res: [AnyStruct] = []
+        for item in array {
             res.append(f(item))
         }
         return res
```
Comment (bluesign) [inline contracts/ArrayUtils.cdc:57]: `return array.map<AnyStruct>(f)`

File: contracts/ArrayUtils.cdc (hunk 1 above)
Comment (turbolent) [reply contracts/ArrayUtils.cdc]: Good point! I didn't realize we had that now, too 👌 Addresses in 8fbd997ddf42d464e0a06df922547c742b841265

File: contracts/ArrayUtils.cdc (hunk 2 above)
Comment (turbolent) [reply contracts/ArrayUtils.cdc]: Good suggestion! Addressed here in `ArrayUtils.map` and also in `ArrayUtils.mapStrings` in 8fbd997ddf42d464e0a06df922547c742b841265

File: contracts/AddressUtils.cdc [hunk 3]
```diff
@@ -49,7 +50,7 @@ access(all) contract AddressUtils {
         return nil
     }
 
-    access(all) fun isValidAddress(_ input: AnyStruct, forNetwork: String): Bool {
+    access(all) fun isValidAddress(_ input: AnyStruct, forNetwork network: String): Bool {
```
Comment (bjartek) [inline contracts/AddressUtils.cdc:53]:
```suggestion
    access(all) fun isValidAddress(_ input: AnyStruct, network: String): Bool {
```

**Review by bluesign — APPROVED** [review]

File: contracts/AddressUtils.cdc (hunk 3 above)
Comment (turbolent) [reply contracts/AddressUtils.cdc:53]: That would be a breaking change, not sure if you want that?

**Comment by turbolent** [comment]
@bjartek @austinkline Would be great to get this merged, so we can update the [Cadence Source Compatibility Suite](https://github.com/onflow/cadence/tree/master/compat), which currently uses a fork with these changes: https://github.com/onflow/cadence/blob/master/compat/suite/green-goo-dao-flow-utils.yaml

**Review by austinkline — APPROVED** [review]
LGTM, @bjartek any objections?

**Comment by bjartek** [comment]
No

**Pull request merged**